### PR TITLE
[SWE-543] Merge changes from main into main-FSS2

### PR DESCRIPTION
**Pull request by tyler-foster**
#### Changes
This PR ports several tickets from `main` over to `main-FSS2`, so that we'll be able to more easily move back down to only have one main branch in the coming future.

The tickets in question, [per the big ticket](https://aicsjira.corp.alleninstitute.org/browse/SWE-543), are:

- [UR-81] Fixed a bug with displaying correct date/datetime values for already-uploaded files 
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/17) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/3f9af31988e9edd6bce78f2b38eac2d8ba4e4dc7)
- [SWE-166] Swapped to using GH Pages for our download page rather than Confluence
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/21) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/a6fa6e2b5f8e87d17de53dd555b00e6d93e696d8)
- [SWE-177] Added tutorials for users. With pictures!
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/23) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/e75329611a60f76850dd899472365e7968820219)
- [SWE-216] Removed the notion of `ImageModel`   <------ This had the most conflicting changes
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/25) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/de1bb8090b01997e702e5b84dfdeacf416be6ac2)
- [SWE-279] Added filenames alongside file IDs in file dropdowns
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/30) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/9ff5c0792b987400f3405a79e60a49e3d79343f4)
- [SWE-335] Improved loadtime on file dropdown searches
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/29) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/cbface4c8f38bb5b7381e4d23ce51bf8f3a016a9)
- [SWE-450] Fixed a bug to enable users to edit files that have been tagged with "unlisted" annotations
  * [Original PR](https://github.com/aics-int/aics-file-upload-app/pull/34) / [Changes](https://github.com/aics-int/aics-file-upload-app/pull/48/commits/b8f896d6c2c23d94b54b96c041f985465f45f363)

All I did for this ticket was run `git merge <merge commit hash>` for the relevant merge commits, and then I manually fixed the (surprisingly few) conflicts. I won't be surprised if it turns out I missed anything

#### How to review this
I'm honestly not sure what the best way to review this would be. The git history looks kind of messy, but if you scroll down to the bottom of the commit log for this PR you'll find individual commits for each ticket merge. Clicking through to the diffs of each one of those could be a good start. It will probably be a good idea for us to ask some users to pilot test a build with these changes as well.

### CODE REVIEW

**Comment by pgarrison** [comment]
My review progress

* [x] [UR-81] Fixed a bug with displaying correct date/datetime values for already-uploaded files
  - LGTM
* [ ] [SWE-166] Swapped to using GH Pages for our download page rather than Confluence
* [ ] [SWE-177] Added tutorials for users. With pictures!
* [ ] [SWE-216] Removed the notion of ImageModel <------ This had the most conflicting changes
* [ ] [SWE-279] Added filenames alongside file IDs in file dropdowns
* [ ] [SWE-335] Improved loadtime on file dropdown searches
* [ ] [SWE-450] Fixed a bug to enable users to edit files that have been tagged with "unlisted" annotations

File: src/renderer/containers/Table/CustomCells/StatusCell/index.tsx [hunk 1]
```diff
@@ -60,12 +60,13 @@ export default function StatusCell(props: CellProps<UploadSummaryTableRow>) {
...
     } else {
-      // Switch the incomplete status tooltip depending on if the post upload
-      // has yet to run vs if it has failed
-      if (etlProcess?.status === JSSJobStatus.FAILED) {
-        tooltip = `${tooltip} - File has been successfully uploaded to FMS, but may not be viewable in the FMS File Explorer. Attempt to make it visible in the FMS Explorer resulted in the following error: ${etlProcess.status_detail}`;
+      if (
+        props.row.original.serviceFields?.postUploadProcessing?.etl?.status ===
+        JSSJobStatus.FAILED
+      ) {
+        tooltip = `${tooltip} - File has been successfully uploaded to FMS, but may not be viewable in the File Upload App. Attempt to make it visible in the FMS Explorer resulted in the following error: ${props.row.original.serviceFields?.postUploadProcessing?.etl?.status_detail}`;
```
Comment (pgarrison) [inline src/renderer/containers/Table/CustomCells/StatusCell/index.tsx:67]: I am suspicious of the changes from `etlProcess?.status` to `props.row.original.serviceFields?.postUploadProcessing?.etl?.status` on lines 64 and 67. In the [original PR](https://github.com/aics-int/aics-file-upload-app/pull/21/files#diff-739a3322a5484e0db015222c5f4f4b972756685a891a911024447a2cedb3bac6) for these changes, the change was simply replacing "FMS File Explorer" with "File Upload App". So I suspect that `etlProcess?.status` is the newer/cleaner option

File: src/renderer/state/route/test/logics.test.ts [hunk 2]
```diff
@@ -337,7 +338,21 @@ describe("Route logics", () => {
         VIEW_UPLOADS_SUCCEEDED
       );
     });
+    it("allows users to open an upload with unexposed annotations", async () => {
+      stubMethods();
```
Comment (pgarrison) [inline src/renderer/state/route/test/logics.test.ts:342]: Note: this was `stubMethods({});` in the [original PR](https://github.com/aics-int/aics-file-upload-app/pull/34/files#), but I'm guessing is correct as `stubMethods();`

File: src/renderer/state/test/mocks.ts [hunk 3]
```diff
@@ -460,8 +496,38 @@ export const mockWorkingUploadJob: UploadJob = {
 };
 
 export const mockWaitingUploadJob: UploadJob = {
-  ...mockWorkingUploadJob,
+  created: new Date(),
+  currentStage: "Waiting",
+  jobId: "2222222222",
+  jobName: "mockWaitingUploadJob",
```
Comment (pgarrison) [inline src/renderer/state/test/mocks.ts:502]: I'm having trouble finding what previous commit this mock code (lines 499 to 530) came from. The [original PR](https://github.com/aics-int/aics-file-upload-app/pull/34/files#diff-f028817c73f3e18d69f2fe44e8c75c213f6ab26dd9f8e587f5c12528b63ac0f6) just added the `mockSuccessfulUploadJobWithUnexposedAnnotation` mock and left this one untouched as follows:
```javascript
export const mockWaitingUploadJob: JSSJob = {
  ...mockWorkingUploadJob,
  currentStage: UploadStage.WAITING_FOR_CLIENT_COPY,
  status: JSSJobStatus.WAITING,
};
```
(Note though that `main-FSS2` doesn't have the `currentStage` line.)

File: src/renderer/state/route/logics.ts [hunk 4]
```diff
@@ -164,98 +160,90 @@ function convertUploadRequestsToUploadStateBranch(
...
+            case ColumnType.DATETIME:
+              values = values.map((v) => new Date(`${v}`));
+              break;
+            case ColumnType.LOOKUP:
+              if (
+                  annotationIdToLookupMap[annotation.annotationId]?.[
+                      "scalarTypeId/Name"
+                      ] === ScalarType.INT
+              ) {
+                values = values.map((v) => parseInt(v, 10));
```
Comment (pgarrison) [inline src/renderer/state/route/logics.ts:197]: The changes in this file look like a lot, but it's actually mostly whitespace. See [the diff ignoring whitespace here](https://gist.github.com/pgarrison/c915c847ddef2b331f5cb565ad190e82)

File: LICENSE.txt [hunk 5]
```diff
@@ -1,13 +1,33 @@
-Allen Institute Software License – This software license is the 2-clause BSD license plus clause a third clause that prohibits redistribution and use for commercial purposes without further permission.
+Allen Institute Software License – This software license is the 2-clause BSD
+license plus a third clause that prohibits redistribution and use for
+commercial purposes without further permission.
 
-Copyright © 2019. Allen Institute. All rights reserved.
+Copyright © 2021. Allen Institute.  All rights reserved.
```
Comment (kmitcham) [inline LICENSE.txt:5]: 2021?  not 2023?

File: src/renderer/containers/Table/CustomCells/StatusCell/index.tsx [hunk 6]
```diff
@@ -60,12 +60,13 @@ export default function StatusCell(props: CellProps<UploadSummaryTableRow>) {
...
       } else {
-        tooltip = `${tooltip} - File has been successfully uploaded; working on making it visible in the FMS File Explorer if it isn't already`;
+        tooltip = `${tooltip} - File has been successfully uploaded; working on making it visible in the File Upload App if it isn't already`;
```
Comment (kmitcham) [inline src/renderer/containers/Table/CustomCells/StatusCell/index.tsx:69]: The File Upload App is making an uploaded file visible in the File Upload App?

**Review by SeanLeRoy — COMMENTED** [review]
Looking great Tyler! Were you able to verify all the features included in the tickets listed are working in this FSS2 world now?

**Review by hughes036 — COMMENTED** [review]
So to create this, you: 

- Branched from `feature/SWE-543_merge_main_to_fss2` off from `main`
- Merged `main-FSS2` into `feature/SWE-543_merge_main_to_fss2` 
- Ran and tested `feature/SWE-543_merge_main_to_fss2`
- Made this pr (`feature/SWE-543_merge_main_to_fss2` -> main-fss2) without any further commits.

Did i get that right?  So if I run and test `feature/SWE-543_merge_main_to_fss2`, I can say that I am testing this diff?

**Comment by pgarrison** [comment]
> So to create this, you:
> 
> * Branched from `feature/SWE-543_merge_main_to_fss2` off from `main`
> * Merged `main-FSS2` into `feature/SWE-543_merge_main_to_fss2`
> * Ran and tested `feature/SWE-543_merge_main_to_fss2`
> * Made this pr (`feature/SWE-543_merge_main_to_fss2` -> main-fss2) without any further commits.
> 
> Did i get that right?

From the commit logs, it looks like Tyler did the following:

* Created the branch `feature/SWE-543_merge_main_to_fss2` from `main-FSS2`
* For each of the 7 PRs on `main` but not on `main-FSS2`, Tyler merged the corresponding commit in `main` into `feature/SWE-543_merge_main_to_fss2` (you can see the 7 merge commits near the top [here](https://github.com/aics-int/aics-file-upload-app/commits/feature/SWE-543_merge_main_to_fss2))
* Added 4 more commits to `feature/SWE-543_merge_main_to_fss2` to fix the `yarn.lock` file
* Ran and tested? I can't speak to that ;)

> So if I run and test `feature/SWE-543_merge_main_to_fss2`, I can say that I am testing this diff?

Partially, but not quite. In the mean time, `main-FSS2` has seen 12 new commits, so ideally we would test the result of merging this PR into `main-FSS2`. So that would look like
* Check out the latest `main-FSS2`
* Merge `feature/SWE-543_merge_main_to_fss2` into `main-FSS2`
* Run & test

**Comment by hughes036** [comment]
> > So to create this, you:
> > 
> > * Branched from `feature/SWE-543_merge_main_to_fss2` off from `main`
> > * Merged `main-FSS2` into `feature/SWE-543_merge_main_to_fss2`
> > * Ran and tested `feature/SWE-543_merge_main_to_fss2`
> > * Made this pr (`feature/SWE-543_merge_main_to_fss2` -> main-fss2) without any further commits.
> > 
> > Did i get that right?
> 
> From the commit logs, it looks like Tyler did the following:
> 
> * Created the branch `feature/SWE-543_merge_main_to_fss2` from `main-FSS2`
> * For each of the 7 PRs on `main` but not on `main-FSS2`, Tyler merged the corresponding commit in `main` into `feature/SWE-543_merge_main_to_fss2` (you can see the 7 merge commits near the top [here](https://github.com/aics-int/aics-file-upload-app/commits/feature/SWE-543_merge_main_to_fss2))
> * Added 4 more commits to `feature/SWE-543_merge_main_to_fss2` to fix the `yarn.lock` file
> * Ran and tested? I can't speak to that ;)
> 
> > So if I run and test `feature/SWE-543_merge_main_to_fss2`, I can say that I am testing this diff?
> 
> Partially, but not quite. In the mean time, `main-FSS2` has seen 12 new commits, so ideally we would test the result of merging this PR into `main-FSS2`. So that would look like
> 
> * Check out the latest `main-FSS2`
> * Merge `feature/SWE-543_merge_main_to_fss2` into `main-FSS2`
> * Run & test

Thanks @pgarrison .  I followed your steps, manually tested again and all passed.

**Review by hughes036 — APPROVED** [review]

**Comment by hughes036** [comment]
Whats the status on this?

**Comment by hughes036** [comment]
Again, any update here?  Is this waiting on something @tyler-foster ?

File: src/renderer/containers/Table/CustomCells/StatusCell/index.tsx (hunk 6 above)
Comment (tyler-foster) [reply src/renderer/containers/Table/CustomCells/StatusCell/index.tsx:69]: Well it _does_ do that, it does make uploaded files visible in the FUA itself after upload, but looking at this again I'm honestly not sure which is more correct for this message...

File: src/renderer/state/test/mocks.ts (hunk 3 above)
Comment (tyler-foster) [reply src/renderer/state/test/mocks.ts:502]: That would track - I believe this was the one test I had to edit manually to get it to play nice with a typing change.

**Comment by tyler-foster** [comment]
> From the commit logs, it looks like Tyler did the following:

:point_up: Everything Phillip said in this comment was correct.

> Again, any update here? Is this waiting on something @tyler-foster ? Or just a second approval?

@hughes036 Second approval, me not being out 2 days a week for the last two weeks, me not being focused on that darned little MNS replacement, and an attempt at re-creating the 409 error I got when trying to upload from this branch the first time (which I've failed to do ((which is a good thing)), it was probably due to some user-error on my part).

> Looking great Tyler! Were you able to verify all the features included in the tickets listed are working in this FSS2 world now?

@SeanLeRoy To the extent of my understanding, yes, I was able to verify all the features included in the tickets work in the FSS2 world now. The date/datetime bug looks correct, the file dropdown improvements were easy to verify, and the GH Pages / tutorials were virtually conflict-less. All of the unit tests are passing with minimal adjustments - do we have faith in those? I'm assuming they're mostly correct.

The only change I wasn't really sure about was the one regarding `ImageModel`, mainly because I just didn't go into that with a ton of context. It appears to be fine though (?) given that it's mostly just removing stuff.

**Review by SeanLeRoy — APPROVED** [review]

**Pull request merged**